packages/react-router-dom/index.tsx

```diff
@@ -616,7 +616,11 @@ export function RouterProvider({
                 navigator={navigator}
               >
                 {state.initialized ? (
-                  <DataRoutes routes={router.routes} state={state} />
+                  <DataRoutes
+                    routes={router.routes}
+                    future={router.future}
+                    state={state}
+                  />
```

Comment:
Plumb down the `router.future` from `RouterProvider` so we can use it in `renderMatches` and only do the fallback stuff when enabled